hasher-matcher-actioner/hashing-data/main.tf

```diff
@@ -0,0 +1,222 @@
+    topic_arn     = aws_sns_topic.image_notification_topic.arn
+    events        = ["s3:ObjectCreated:*"]
+    filter_prefix = "images/"
+  }
+
+  topic {
+    topic_arn     = aws_sns_topic.threat_exchange_data.arn
+    events        = ["s3:ObjectCreated:*"]
+    filter_prefix = "threat_exchange_data/"
+  }
+}
+
+
+# ThreatExchange Data File Folder
+
+resource "aws_s3_bucket_object" "threat_exchange_data" {
+  bucket       = aws_s3_bucket.data_bucket.id
+  key          = "threat_exchange_data/"
+  content_type = "application/x-directory"
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "ThreatExchangeDataFolder"
+    }
+  )
+}
+
+resource "aws_sns_topic" "threat_exchange_data" {
+  name_prefix = "${var.prefix}-threatexchange-data"
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "ThreatExchangeDataFolderUpdated"
+    }
+  )
+}
+
+data "aws_iam_policy_document" "threat_exchange_data" {
+  statement {
+    effect    = "Allow"
+    actions   = ["SNS:Publish"]
+    resources = [aws_sns_topic.threat_exchange_data.arn]
+    principals {
+      type        = "Service"
+      identifiers = ["s3.amazonaws.com"]
+    }
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = [aws_s3_bucket.data_bucket.arn]
+    }
+  }
+}
+
+resource "aws_sns_topic_policy" "threat_exchange_data" {
+  arn    = aws_sns_topic.threat_exchange_data.arn
+  policy = data.aws_iam_policy_document.threat_exchange_data.json
+}
+
+
+# Index File Folder
+
+resource "aws_s3_bucket_object" "index" {
+  bucket       = aws_s3_bucket.data_bucket.id
+  key          = "index/"
+  content_type = "application/x-directory"
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "IndexesFolder"
+    }
+  )
+}
+
+# Image File Notifications
+
+resource "aws_s3_bucket_object" "images" {
+  bucket       = aws_s3_bucket.data_bucket.id
+  key          = "images/"
+  content_type = "application/x-directory"
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "ImagesContentFolder"
+    }
+  )
+}
+
+resource "aws_sns_topic" "image_notification_topic" {
+  name_prefix = "${var.prefix}-images"
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "ImagesContentFolderUpdated"
+    }
+  )
+}
+
+data "aws_iam_policy_document" "image_notification_topic_policy" {
+  statement {
+    effect    = "Allow"
+    actions   = ["SNS:Publish"]
+    resources = [aws_sns_topic.image_notification_topic.arn]
+    principals {
+      type        = "Service"
+      identifiers = ["s3.amazonaws.com"]
+    }
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = [aws_s3_bucket.data_bucket.arn]
+    }
+  }
+}
+
+resource "aws_sns_topic_policy" "image_notification_topic_policy" {
+  arn = aws_sns_topic.image_notification_topic.arn
+
+  policy = data.aws_iam_policy_document.image_notification_topic_policy.json
+}
+
+# DynamoDB Datastore
+
+resource "aws_dynamodb_table" "datastore" {
```

Comment:
Is it worth documenting here or elsewhere what these keys are intended to be associated with? I think one would be able to figure it out by reviewing the lambda code but it's a little confusing.

Reply:
Probably not here in the terraform code, but we should probably document this in the overall design documentation, which we should publish here in the repo when we are closer to a release.